packages/components/src/FormField/FormField.tsx

```diff
@@ -1,426 +1,216 @@
+            onFocus: handleFocus,
+            onKeyDown: handleKeyDown,
+          };
+
+          return renderField();
+
+          function renderField() {
+            switch (type) {
+              case "select":
+                return (
+                  <>
+                    <select {...fieldProps}>{children}</select>
+                    <span className={styles.postfix}>
+                      <Icon name="arrowDown" />
+                    </span>
+                  </>
+                );
+              case "textarea":
+                return (
+                  <textarea
+                    rows={rows}
+                    ref={inputRef as MutableRefObject<HTMLTextAreaElement>}
+                    {...textFieldProps}
+                  />
+                );
+              default:
+                return (
+                  <>
+                    <input
+                      autoComplete={setAutocomplete(autocomplete)}
+                      type={type}
+                      maxLength={maxLength}
+                      max={max}
+                      min={min}
+                      ref={inputRef as MutableRefObject<HTMLInputElement>}
+                      {...textFieldProps}
+                    />
+                    {loading && <FormSpinner />}
+                  </>
+                );
+            }
+          }
+
+          function handleChange(
+            event: ChangeEvent<
+              HTMLInputElement | HTMLTextAreaElement | HTMLSelectElement
+            >,
+          ) {
+            let newValue: string | number;
+            newValue = event.currentTarget.value;
+
+            if (type === "number" && newValue.length > 0) {
+              newValue = parseFloat(newValue);
+            }
+
+            setHasMiniLabel(String(newValue).length > 0);
+            onChange && onChange(newValue);
+            onControllerChange(event);
+          }
+
+          function handleKeyDown(
+            event: KeyboardEvent<HTMLInputElement | HTMLTextAreaElement>,
+          ) {
+            if (!onEnter) return;
+            if (event.key !== "Enter") return;
+            if (event.shiftKey || event.ctrlKey) return;
+
+            event.preventDefault();
+            onEnter && onEnter(event);
+          }
+
+          function handleFocus(
+            event: FocusEvent<HTMLInputElement | HTMLTextAreaElement>,
+          ) {
+            const target = event.currentTarget;
+            setTimeout(() => readonly && target.select());
+
+            onFocus && onFocus();
+          }
+
+          function handleBlur() {
+            onBlur && onBlur();
+            onControllerBlur();
+          }
+
+          function setAutocomplete(autocompleteSetting: boolean | string) {
+            if (autocompleteSetting === "one-time-code") {
+              return "one-time-code";
+            }
+
+            return autocompleteSetting ? undefined : "autocomplete-off";
+          }
+        }}
+      />
```

Comment:
The jsx here is long and it's kind of using the renderProps technique that was kind of replaced by hooks. Feel free to take a look at here and see how we do it on the mobile side:
https://github.com/GetJobber/jobber-mobile/pull/952/files

Reply:
@hosseintalebi yeah, I agree. I tried using the hook at first and struggled with it. To be honest I don't really remember why. 

One thing to note is that we found some typing issues on `v6` of react hook forms that could have been where I got stuck.

Do you think it is worth investigating using the hook?

Reply:
@eddysims We have made some progress on the mobile side to only use the hooks. Let's get back to it after we have the mobile side ready.